scripts/lib/libvirt/libvirt_setup.py

```diff
@@ -363,7 +386,7 @@ def cleanup(args):
         domain_cleanup(dom)
 
     for network in conn.listAllNetworks():
-        if network.name() == args.cloud + "-admin":
+        if network.name() in (args.cloud + "-admin", args.cloud + "-ironic"):
```

Comment:
Well... there was one network harcoded before :)
As this function is running over all networks and just picking the ones to clean up, I think that this is a good solution. Note that we want to cleanup cloud-ironic network even if it was not planned to be created for current deployment to remove potential leftovers from previous runs.

The only improvement I see here would be to not du `args.cloud` for each name but rather strip it from the `network.name()` and compare the suffixes.